From: "Miguel A. Vico" <mvicomoya@nvidia.com>
Date: Wed, 25 Sep 2019 11:23:13 -0700
Subject: desktop-shell: Avoid NULL output dereference when getting surface
 label

When hotunplugging a display, the compositor will tear down the
corresponding output object.

Avoid NULL output dereferences by all surface label getters in
desktop-shell when hotunplugging happens.

Signed-off-by: Christopher Obbard <chris.obbard@collabora.com>
(cherry picked from commit 620f68dc4f72217b5992a344d38b116fe771ce0b)
---
 desktop-shell/shell.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index f94a4ca..00ee107 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -521,7 +521,7 @@ static int
 focus_surface_get_label(struct weston_surface *surface, char *buf, size_t len)
 {
 	return snprintf(buf, len, "focus highlight effect for output %s",
-			surface->output->name);
+			(surface->output ? surface->output->name : "NULL"));
 }
 
 /* no-op func for checking focus surface */
@@ -2985,7 +2985,7 @@ static int
 background_get_label(struct weston_surface *surface, char *buf, size_t len)
 {
 	return snprintf(buf, len, "background for output %s",
-			surface->output->name);
+			(surface->output ? surface->output->name : "NULL"));
 }
 
 static void
@@ -3065,7 +3065,7 @@ static int
 panel_get_label(struct weston_surface *surface, char *buf, size_t len)
 {
 	return snprintf(buf, len, "panel for output %s",
-			surface->output->name);
+			(surface->output ? surface->output->name : "NULL"));
 }
 
 static void
